java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/docker/DefaultDockerfileGenerator.java

```diff
@@ -30,56 +29,55 @@
  */
 public class DefaultDockerfileGenerator implements DockerfileGenerator {
 
-  private static final String DOCKERFILE = "FROM %s\n"
-      + "ADD %s %s\n";
+  private static final String DOCKERFILE_FROM = "FROM %s\n";
+  private static final String DOCKERFILE_ADD_ARTIFACT = "ADD %s %s\n";
+  private static final String DOCKERFILE_JETTY_QUICKSTART = "RUN /scripts/jetty/quickstart.sh\n";
+
+  private static final String APP_DESTINATION = "$APP_DEST";
 
-  private final String jarRuntime;
-  private final String serverRuntime;
-  private final String tomcatRuntime;
+  private final JdkServerLookup runtimeLookupTool;
 
   /**
    * Constructs a new {@link DefaultDockerfileGenerator}.
    */
   @Inject
-  DefaultDockerfileGenerator(@JarRuntimeImage String jarRuntime,
-      @ServerRuntimeImage String serverRuntime,
-      @TomcatRuntimeImage String tomcatRuntime) {
-    this.jarRuntime = jarRuntime;
-    this.serverRuntime = serverRuntime;
-    this.tomcatRuntime = tomcatRuntime;
+  DefaultDockerfileGenerator(JdkServerLookup runtimeLookupTool) {
+    this.runtimeLookupTool = runtimeLookupTool;
   }
 
   @Override
   public String generateDockerfile(Path artifactToDeploy, RuntimeConfig runtimeConfig) {
     StringBuilder dockerfile = new StringBuilder();
     String fileType = Files.getFileExtension(artifactToDeploy.toString());
     String baseImage;
-    String appDest;
 
-    if (fileType.equals("jar")) {
-      baseImage = jarRuntime;
-      appDest = "app.jar";
-    } else if (fileType.equals("war")) {
-      if (runtimeConfig.getServer() != null
-          && runtimeConfig.getServer().equals("tomcat")) {
-        baseImage = tomcatRuntime;
-        appDest = "ROOT.war";
-      } else {
-        baseImage = serverRuntime;
-        appDest = "$JETTY_BASE/webapps/root.war";
-      }
+    if (fileType.equalsIgnoreCase("jar")) {
```

Comment:
We might want to prioritize the user input (app.yaml) rather than the artifact type, this could be useful for example to support exploded war or deployment of multiple war.

However this could be addressed in a different PR as it would introduce breaking change.